modules/core/shared/src/main/scala/weaver/suites.scala

```diff
@@ -60,7 +59,6 @@ object EffectSuite {
 
 }
 
-@RunWith(classOf[weaver.junit.WeaverRunner])
```

Comment:
Mmm this annotation has a pretty significant importance (despite the fact that there's no test associated to it): it's what allows IntelliJ to run test suites by clicking on them.

Reply:
I thought it was important, but wanted to get to a green build before fixing it.

Our doc generation fails in Scala 3:
```
sbt:cats-core> doc
[error] -- Error: modules/core/shared/src/main/scala/weaver/suites.scala:63:0 ----------
[error] 63 |@RunWith(classOf[weaver.junit.WeaverRunner])
[error]    |^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |undefined: new org.junit.runner.RunWith # -1: TermRef(TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class junit)),object runner),RunWith),<init>) at readTasty
[error] one error found
```

I haven't dug deep into it yet, but it seems like a bug in dottydoc or the tasty reader. Possibly, it can't read the the tasty generated by `suites.scala`.

Reply:
It's probably because the dependency is tagged as Optional in this module. 

We could make the dependency required in the scope of the task that requires it maybe ? Not gonna lie, my SBT skills are not good enough to know how to do that from the top of my head

Reply:
That seems to be the problem.
 - `junit` is an `Optional` dependency of `core`. 
 - `coreJVM / doc` has the junit jar on the classpath.
 - `coreCats` depends on `core`. However `coreCatsJVM / doc` doesn't have the junit jar.

The classpath rules are the same for Scala 2.13 and 3, but dottydoc uses the tasty reader, which seems to need all jars to be present.

We can work around this by adding the `junit` dependency with a `ScalaDocTool` configuration.

Reply:
that's great, thank you !